scripts/contiv-vpp-bug-report.sh

```diff
@@ -104,13 +104,66 @@ get_shell_data_ssh() {
     ssh "$SSH_USER@$NODE_NAME" "${SSH_OPTS[@]}" "$@" > "$log" || true
 }
 
+read_shell_data_local() {
+    "$@" || true
+}
+
+read_shell_data_ssh() {
+    ssh "$SSH_USER@$NODE_NAME" "${SSH_OPTS[@]}" "$@" || true
+}
+
+read_node_shell_data() {
+    if [ "$USE_SSH" = "1" ]
+    then
+        read_shell_data_ssh $@
+    else
+        read_shell_data_local $@
+    fi
+}
+
 get_k8s_data() {
     log="$1"
     shift
     echo " - kubectl $*"
     master_kubectl "$@"> "$log"
 }
 
+save_container_nw_report() {
+    # generates Docker container network report for the $NODE_NAME into the $CONTAINER_NW_REPORT_FILE
+    # for each docker container, prints out its interfaces, routing table and ARP table
+    echo " - ${CONTAINER_NW_REPORT_FILE}"
+
+    # get ID + NAME of each container except the pause containers
+    CONTAINERS_TXT=$(read_node_shell_data "sudo docker ps --format '{{.ID}} {{.Names}}' | grep -v pause")
+    while read -r line; do CONTAINERS+=( "$line" ); done <<< "$CONTAINERS_TXT"
+
+    # return in case of no container data (e.g. issues with executing sudo)
+    if [ -z "${CONTAINERS}" ]; then
+        return
+    fi
+
+    # loop over each container
+    for CONTAINER_DATA in "${CONTAINERS[@]}"
+    do
+        # split CONTAINER_DATA to container ID and container Name
+        IFS=' ' read -ra CINFO <<< "$CONTAINER_DATA"
```

Comment:
Should use lower case local variables.  Also, it might be better to use individual variables here instead of reading into an array.  That way if there are less elements than you expect, you won't get an undefined variable error below.